buildSrc/src/main/java/Publication.kt

```diff
@@ -1,228 +1,47 @@
-            renameModule(project.name, "tvosx64", version)
-        }
-        from("build${sep}libs$sep") {
-            include("${project.name}-tvosx64*")
-            withJavadoc(project.name, "tvosx64")
-        }
-        fromKlib(project.name, "tvosX64", version)
-    }
-    this.maybeCreate("tvossimulatorarm64").contents {
-        from("build${sep}publications${sep}tvosSimulatorArm64") {
-            renameModule(project.name, "tvossimulatorarm64", version)
-        }
-        from("build${sep}libs$sep") {
-            include("${project.name}-tvossimulatorarm64*")
-            withJavadoc(project.name, "tvossimulatorarm64")
-        }
-        fromKlib(project.name, "tvosSimulatorArm64", version)
-    }
-}
+fun DistributionContainer.configureForMultiplatform(project: Project, buildPublishDir: String) {
+    val version = project.property("versionName").toString()
+    if (version.isEmpty()) {
+        throw GradleException("DistZip: version name is empty")
+    }
+    val projectName = project.name
+    val platforms = mapOf(
+        "main" to projectName,
+        "android" to "$projectName-android",
+        "jvm" to "$projectName-jvm",
+        "iosarm64" to "$projectName-iosarm64",
+        "iossimulatorarm64" to "$projectName-iossimulatorarm64",
+        "iosx64" to "$projectName-iosx64",
+        "macosarm64" to "$projectName-macosarm64",
+        "macosx64" to "$projectName-macosx64",
+        "tvosarm64" to "$projectName-tvosarm64",
+        "tvossimulatorarm64" to "$projectName-tvossimulatorarm64",
+        "tvosx64" to "$projectName-tvosx64",
+        "watchosarm32" to "$projectName-watchosarm32",
+        "watchosarm64" to "$projectName-watchosarm64",
+        "watchossimulatorarm64" to "$projectName-watchossimulatorarm64",
+        "watchosx64" to "$projectName-watchosx64"
+    )
 
-private fun CopySpec.fromKlib(projectName: String, target: String, version: String) {
-    val pos = projectName.length
-    from("build${sep}classes${sep}kotlin${sep}${target}${sep}main${sep}cinterop") {
-        include("*.klib")
-        rename {
-            it.replaceRange(pos, pos, "-${target.lowercase()}-$version")
-        }
-    }
-    from("build${sep}classes${sep}kotlin${sep}${target}${sep}main${sep}klib") {
-        rename {
-            "$projectName-${target.lowercase()}-$version.klib"
-        }
-    }
-}
+    platforms.forEach { (distName, projectName) ->
+        val distribution = if (distName == "main") getByName("main") else maybeCreate(distName)
+        distribution.contents {
+            val basePath = "${buildPublishDir}io${sep}sentry${sep}$projectName$sep$version"
 
-private fun CopySpec.renameModule(projectName: String, renameTo: String = "", version: String) {
-    var target = ""
-    if (renameTo.isNotEmpty()) {
-        target = "-$renameTo"
-    }
-    rename {
-        it.replace("module.json", "$projectName$target-$version.module")
-    }
-}
+            // Rename the POM since craft looks for pom-default.xml
```

Comment:
ugh, I guess ideal-world we wouldn't need this, but would have to change craft again 🙈 maybe makes sense to align this between repos and change craft at some point